common/source/kernel/Condition.cpp

```diff
@@ -46,6 +46,10 @@ void Condition::wait(bool re_acquire_mutex, pointer called_by)
           "This may lower the performance of the system, and cause undetectable deadlocks!\n",
           currentThread->getName(), currentThread, getName(), this);
     printHoldingList(currentThread);
+
+    // This might not be a problem, but it is slow and WILL lead to hard-to-find deadlocks in many cases.
+    // If you want to remove this assert, talk to your tutor FIRST!
+    assert(!currentThread->holding_lock_list_->hasNextOnHoldingList() && "thread is holding locks while waiting on a condition variable");
```

Comment:
this is saying, holding any lock except for the one the CV needs? could the comment reflect that? otherwise students might think they have to unlock the mutex before waiting....

Reply:
It should be more clear now.